rust/operator/src/hdfs_controller.rs

```diff
@@ -396,14 +498,78 @@ fn rolegroup_statefulset(
+                        name: Some(rolegroup_ref.object_name()),
+                        ..ConfigMapVolumeSource::default()
+                    }),
+                    ..Volume::default()
+                },
+                Volume {
+                    name: "kerberos".to_string(),
+                    csi: Some(CSIVolumeSource {
+                        driver: "secrets.stackable.tech".to_string(),
+                        volume_attributes: Some(
+                            [
+                                (
+                                    "secrets.stackable.tech/class".to_string(),
+                                    "kerberos".to_string(),
+                                ),
+                                (
+                                    "secrets.stackable.tech/scope".to_string(),
+                                    "pod,node".to_string(),
+                                ),
+                                (
+                                    "secrets.stackable.tech/kerberos.service.names".to_string(),
+                                    match role {
+                                        HdfsRole::JournalNode => "jn,HTTP",
+                                        HdfsRole::NameNode => "nn,HTTP",
+                                        HdfsRole::DataNode => "dn,HTTP",
+                                    }
+                                    .to_string(),
+                                ),
+                            ]
+                            .into(),
+                        ),
+                        ..CSIVolumeSource::default()
+                    }),
+                    ..Volume::default()
+                },
+                Volume {
+                    name: "tls".to_string(),
+                    csi: Some(CSIVolumeSource {
+                        driver: "secrets.stackable.tech".to_string(),
+                        volume_attributes: Some(
+                            [
+                                (
+                                    "secrets.stackable.tech/class".to_string(),
+                                    "tls".to_string(),
+                                ),
+                                (
+                                    "secrets.stackable.tech/scope".to_string(),
+                                    "pod,node".to_string(),
+                                ),
+                            ]
+                            .into(),
+                        ),
+                        ..CSIVolumeSource::default()
+                    }),
+                    ..Volume::default()
+                },
+                Volume {
+                    name: "keystore".to_string(),
+                    empty_dir: Some(EmptyDirVolumeSource::default()),
+                    ..Volume::default()
+                },
+            ]),
+            security_context: Some(PodSecurityContext {
+                fs_group: Some(9999),
```

Comment:
In other operators we use https://github.com/stackabletech/trino-operator/blob/d143a7111cfab52fe75bf997ca00b876005e55b0/rust/operator-binary/src/controller.rs#L901-L906
```
        .security_context(PodSecurityContext {
            run_as_user: Some(1000),
            run_as_group: Some(1000),
            fs_group: Some(1000),
            ..PodSecurityContext::default()
        });
```
Not saying that's better, just for consistency reasons